### PR TITLE
2018-02-09 authentication bug

### PR DESCRIPTION
**Pull request by djmitche**

### CODE REVIEW

File: retrospectives/2018-02-09-authentication-bug.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+#### Bugzilla Bug: [bug 1437131](https://bugzilla.mozilla.org/show_bug.cgi?id=1437131)
+
+## Summary
+
+As a result of a software bug, HTTP requests to many Taskcluster services that resulted a 401 response *achieved their side-effect anyway*.
+
+### Background
+
+HTTP requests to Taskcluster APIs are authenticated using [Hawk](https://github.com/hueniverse/hawk),
+The authentication process determines the [client](https://docs.taskcluster.net/manual/design/apis/hawk/clients) making the request, and that cloent's [scopes](https://docs.taskcluster.net/manual/design/apis/hawk/scopes) are uesd to determine whether the request can proceed.
```
Comment (imbstack) [inline retrospectives/2018-02-09-authentication-bug.md]: cloent

File: retrospectives/2018-02-09-authentication-bug.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+The bug was tracked down quickly and fixed with a [short patch](https://github.com/taskcluster/taskcluster-lib-api/commit/488d9fd7e9569bb464039778409d50ac88a13f8e).
+To avoid drawing attention to the issue, we chose to release the fix as a package on npmjs.com until all services were patched, only pushing to Github once the danger had passed.
+The fix was quickly deployed to all of the affected services.
+
+The bug was introduced in `taskcluster-lib-api` version 5.0.0 and fixed in version 6.0.4.
+Version 5.0.0 was not widely deployed, and version 6.0.0 was released soon after.
+
+### Remediation
+
+We have no reason to think that this bug was exploited, but we used two approaches to verify this.
```
Comment (imbstack) [inline retrospectives/2018-02-09-authentication-bug.md]: I don't think is is "but" right? More like "and" or semicolon.

**Review by imbstack — APPROVED** [review]
Really great job on this.

**Review by jonasfj — APPROVED** [review]

**Review by ccooper — APPROVED** [review]
LGTM

**Pull request merged**